### DISCUSSION TITLE
training  KeywordRecognizer

### DISCUSSION THREAD

**Discussion by ramkirank** [spokestack/spokestack-python#102] [Q&A]
Can  spokestack-python be used independently to train KeywordRecognizer model(or SpokestackKeyword:) . I want to use keras  model.fit() or tensorflow training loop.

**Comment by will-rice** [comment]
The python library does not allow you to directly train models at this time.

If you would like to make a custom model you can do so via Maker with an [account](https://www.spokestack.io/account/create).

In addition, you can also check out the docs for more about our [wakeword](https://www.spokestack.io/docs/machine-learning/wakeword-models) and [keyword](https://www.spokestack.io/docs/concepts/keywords/) models.

**Reply by ramkirank** [reply]
is there any plan to support this feature in future releases? . And can you please suggest where I can find model architecture in detail(like if conv1d or dense layers used in filter/encode/detect.tflite for  KeywordRecognizer or others)

**Reply by will-rice** [reply]
Spokestack is intended as a runtime library so it is unlikely that we would add that feature in the future. Spokestack is intended as a runtime library so it is unlikely that we will support that feature in the future. The keyword model is very similar to the wake word model. The wake word section I linked goes over the basics, but at the bottom of the page, there is a section with references we used in the creation of both models which have a lot more detail.
